text/0000-deprecate-computed-clobberability.md

```diff
@@ -0,0 +1,176 @@
+  }
+
+  let foo = new Foo();
+
+  foo.bar; // undefined
+  foo.bar = 'baz'; // throws an error in strict mode
+}
+```
+
+By constrast, computed properties without setters will be overridden when they
+are set, meaning the computed property is removed from the object and replaced
+with the set value:
+
+```js
+const Foo = EmberObject.extend({
+  bar: computed('_value', {
+    get() {
+      return this._value;
+    },
+  }),
+});
+
+let foo = Foo.create();
+
+foo.bar; // undefined
+foo.set('bar', 'baz'); // Overwrites the getter
+foo.bar; // 'baz'
+foo.set('_value', 123);
+foo.bar; // 'baz'
+```
+
+This behavior is confusing to newcomers, and oftentimes unexpected. Common best
+practice is to opt-out of it by declaring the property as `readOnly`, which
+prevents this overridability.
+
+# Transition Path
+
+This RFC proposes that `readOnly` properties become the default, and that in
+order to override users must opt in by defining their own setters:
+
+```js
+class Foo {
+  get bar() {
+    if (this._bar) {
+      return this._bar;
+    }
+
+    return this._value
+  }
+
+  set bar(value) {
+    this._bar = value
+  }
+}
+```
+
+## Macros
+
+Most computed macros are overridable by default, the exception being `readOnly`.
+This RFC proposes that all computed macros with the exception of `reads` would
+become read only by default. The purpose of `reads` is to _be_ overridable, so
+its behavior would remain the same.
+
+## Decorator Interop
+
+It may be somewhat cumbersome to write overriding functionality or add proxy
+properties when overriding is needed. In an ideal world, computed properties
+would modify accessors transparently so that they could be composed with other
+decorators, such as an `@overridable` decorator:
+
+```js
+class Foo {
+  @overridable
+  @computed('_value')
+  get bar() {
+    return this._value;
+  }
+
+  @overridable
+  @and('baz', 'qux')
+  quux;
+}
+```
+
+Currently this is not possible as computed properties store their getter/setter
+functions elsewhere and replace them with a proxy getter and the mandatory
+setter assertion, respectively. In the long term, making computeds more
+transparent in this way would be ideal, but it is out of scope for this RFC.
+
+## Deprecation Timeline
+
+This change will be a breaking change, which means we will not be able to change
+the behavior of `computed` until Ember v4.0.0. Additionally, users will likely
+want to continue using `.readOnly()` up until overriding has been fully removed
+to ensure they are using properties safely. With that in mind, the ordering of
+events should be:
+
+1. Ember v3
+  * Deprecate the default override-setter behavior immediately. This means that
+    a deprecation warning will be thrown if a user attempts to set a
+    non-`readOnly` property which does not have a setter. User's will still be
+    able to declare a property is `readOnly` without a deprecation warning.
+  * Add optional feature to change the deprecation to an assertion after the
```

Comment:
We chatted about this today in the core team meeting, and decided to try to leverage svelte here instead of an optional feature. This doesn't change the timeline at all here, just the verbiage.

Can you tweak this from `Add optional feature` to `When the application is configured with svelte, the deprecation will be an assertion`?